streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -333,6 +339,43 @@ public static boolean isPermitted(
         return byteArray -> deserializer.deserialize(serdes.topic(), byteArray);
     }
 
+
+    public static Map<TopicPartition, Long> positionToTopicPartitionMap(final Position position) {
+        final Map<TopicPartition, Long> topicPartitions = new HashMap<>();
+        final Set<String> topics = position.getTopics();
+        for (final String t : topics) {
+            final Map<Integer, Long> partitions = position.getPartitionPositions(t);
+            for (final Entry<Integer, Long> e : partitions.entrySet()) {
+                final TopicPartition tp = new TopicPartition(t, e.getKey());
+                topicPartitions.put(tp, e.getValue());
+            }
+        }
+        return topicPartitions;
+    }
+
+    public static void checkpointPosition(final OffsetCheckpoint checkpointFile, final Position position) throws IOException {
+        final Map<TopicPartition, Long> topicPartitions = StoreQueryUtils.positionToTopicPartitionMap(position);
+        checkpointFile.write(topicPartitions);
+    }
+
+    public static Position readPositionFromCheckpoint(final OffsetCheckpoint checkpointFile) {
+        try {
+            final Map<TopicPartition, Long> topicPartitions = checkpointFile.read();
+            return StoreQueryUtils.topicPartitionMapToPosition(topicPartitions);
+        } catch (final IOException e) {
```

Comment:
The write and read paths here handle IOExceptions differently. It would be good for them both to just throw the IOException up to the caller or for them both to catch it and throw a ProcessorStateException.

Reply:
+1